### ISSUE TITLE
Segfault calling "Distinct" on parquet array/list column 

### ISSUE THREAD

**Issue by groceryheist** [duckdb/duckdb#3400]
#### What happens?

I have a parquet dataset that was generating using Spark and has an array column (called `redirect`).  Many queries work fine on this dataset, but when I run queries that include `DISTINCT` on the array column I get a segfault.

Unfortunately, I am struggling to reproduce this on a small dataset.  I can share the full dataset via direct message, but it is too large to post the link publicly.  

#### To Reproduce
Honestly, I am having trouble reproducing this with a smaller dataset.  Download my full dataset (32.7GB uncompressed) and then attempt to run:
 `"SELECT DISTINCT redirect FROM `read_parquet('fandom_2020_wikiq.parquet/*.parquet');")`

#### Environment (please complete the following information):
 - OS: Rocky OS
 - DuckDB Version: 0.3.2
 - DuckDB Client: Python

#### Before Submitting
I have tried this on the latest `master` branch and the same error occurs.

**Comment by hannes** [comment]
Thanks for reporting this, but please try to make a reproducible small example dataset that we could also include in a test case.

**Comment by groceryheist** [comment]
I have tried but taking a small or even moderately sized sample doesn't
seem to replicate the issue. I can work around it and I unfortunately can't
prioritize digging further.

**Comment by groceryheist** [comment]
I managed to find a small dataset that triggers the bug. 
[duckdb_bug.tar.gz](https://github.com/duckdb/duckdb/files/8457647/duckdb_bug.tar.gz)

Extract and run the following to reproduce:
```
import duckdb
df = duckdb.query(f"SELECT DISTINCT redirect FROM read_parquet('test_data.parquet');")
print(df)
```

**Comment by hannes** [comment]
I just tried this with the latest DuckDB pre-release, did not crash for me. Maybe this is already fixed, could you try with the latest pre-release too? `pip install duckdb --upgrade --pre`

**Comment by groceryheist** [comment]
Nope, it still crashes. On Debian 11.3 or Rocky 8.4.

**Comment by groceryheist** [comment]
Ah, does that mean my installations using `--upgrade --pre` did not effectively install the master version?

**Comment by taniabogatsch** [comment]
Yes, I think this is possible. The bug with installing the latest version of DuckDB should be fixed now.

I just ran `pip install duckdb==0.3.3` and I am no longer getting the segmentation fault. Maybe you can give this a try and see if it fixes the problem? :)
